### PR TITLE
Mostly nice-to-have stuff

### PR DESCRIPTION
**Pull request by alex-korobko**
Just can't help making improvements, professional deformation, you now.  But I'll stop at this point.

### CODE REVIEW

File: add_nyt_bestseller_tag.py [hunk 1]
```diff
@@ -125,29 +126,44 @@ def __process_bestseller_group_record(self, bestseller_group_record,
...
+                    job_results['isbns_failed'] + 1
+                self.__save_job_results(job_results)
+                raise
             except:
                 self.logger.exception('Failed to process ISBN {}'
                                       .format(bstslr_record_isbn))
                 job_results['isbns_failed'] = job_results['isbns_failed'] + 1
 
     def run(self) -> None:  # overwrites the AbstractBotJob run method
-        self.dry_run = self.args.dry_run
```
Comment (RayBB) [inline add_nyt_bestseller_tag.py]: In my testing, unless we have this line it will always run in dry run mode

File: add_nyt_review_link.py [hunk 2]
```diff
@@ -162,13 +170,27 @@ def run(self) -> None:  # overwrites the AbstractBotJob run method
...
+            last_shown_on_progress_bar = 0
+            progress_bar_capacity = len(review_record_array)
+            with tqdm(total=progress_bar_capacity, unit='reviews') as pbar:
+                for index, review_record in enumerate(review_record_array):
+                    self.__process_review_record(review_record,
+                                                 comment, job_results)
+                    if index > 0 and index % 25 == 0:
+                        last_shown_on_progress_bar = index
+                        pbar.update(25)
+                pbar.update(progress_bar_capacity - last_shown_on_progress_bar)
```
Comment (RayBB) [inline add_nyt_review_link.py:182]: why manually control tqdm vs use the wrapper around the list?

**Review by RayBB — CHANGES_REQUESTED** [review]

File: add_nyt_review_link.py (hunk 2 above)
Comment (alex-korobko) [reply add_nyt_review_link.py:182]: It doesn't look good, the progress bar printed under every isbn and just makes reading logs harder. The tags bot prints the progress line after every bestseller group, so the progress bar is shown in the logs every 10 ISBNs or so.  That code mimics the same output.

File: add_nyt_bestseller_tag.py (hunk 1 above)
Comment (alex-korobko) [reply add_nyt_bestseller_tag.py]: That's weird. You are right, even if I set the dry-run param explicitly to False it doesn't help and the dry run disclaimer is shown
```
python3 add_nyt_review_link.py --file=./example_responses/reviews.json --dry-run=False
```


Still looking into it and not sure what is going on there.

**Pull request merged**